### DISCUSSION TITLE
Convert PDF to JPG with Imagick - specific PDF not generated properly

### DISCUSSION THREAD

**Discussion by hermans** [ImageMagick/ImageMagick#4234] [Help] [answered]
I'm using Imagick in PHP to convert some PDF files to JPG format, I want to convert 1 page only. But some PDF generated not properly. Here example result:

![image](https://user-images.githubusercontent.com/938767/133610861-e5605395-0d2a-4cc9-b0e2-67feda0c354b.png)

image 3 & 4 is correct, it's not have white space in left. but image 5 & 6 is wrong.

this is my code that I use to do it:
~~~
$im = new Imagick();
        $im->readimage($file . '[0]'); 
        $im->setImageFormat('jpeg'); 
        $im = $im->flattenImages(); // Use this instead            
        $im->setImageCompression(imagick::COMPRESSION_JPEG); 
        $im->setImageCompressionQuality(100); 
        
        //resize 
        $im->resizeImage(300,500, imagick::FILTER_SINC , 1);
        $im->writeImage($dest_file);    
        $im->clear(); 
        $im->destroy();
~~~

**Comment by snibgo** [comment]
Does the IM result look like the image in Adobe Acrobat Reader?

Please post a sample PDF.

Perhaps "-define pdf:use-cropbox=true" or "-define pdf:use-trimbox=true" before reading the PDF will help. See https://imagemagick.org/script/defines.php

**Reply by hermans** [reply]
In Adobe Reader look good, I don't see issue

This is pdf sample: 
[https://drive.google.com/file/d/1xsY8lJ7oo5vKgZRcQG2d0Fqbu9Ksf0qb/view?usp=sharing](url)

**Comment by snibgo** [comment, marked as answer]
Did you try this?
> Perhaps "-define pdf:use-cropbox=true" or "-define pdf:use-trimbox=true" before reading the PDF will help. 

Either of those seems to be what you want.

**Reply by hermans** [reply]
Amazing, thank you :)
